lib/fix/delete_asset_resources.rb

```diff
@@ -3,20 +3,69 @@
 
 module Fix
   class DeleteAssetResources < BatchProcess
+
     def run
-      asset_resources.each do |ar|
-        log.info "Destroying Asset Resource #{ar.id}"
-        begin
-          Hyrax.persister.delete(resource: ar)
-          Hyrax.index_adapter.delete(resource: ar)
-          Hyrax.index_adapter.connection.commit
-          log.info "Asset Resource #{ar.id} destroyed."
-        rescue => e
-          log_error e
-        end
+      log.info "Deleting #{asset_resources.count} Asset Resources..."
+      asset_resources.map do |ar|
+        destroy_work_and_members(resource: ar)
+      end
+      log.info "Done."
+      log.info report
+    end
+
+    private
+
+
+    def results; @results ||= []; end
+
+    def work_destroy_transaction
+      Hyrax::Transactions::WorkDestroy.new
+    end
+
+    def destroy_work_and_members(resource:)
+      result = { resource: resource }
+
+      resource.members.each do |member|
+        destroy_work_and_members(resource: member)
       end
-      puts "Done."
+
+      log.info "DELETING #{resource.class} #{resource.id.id}..."
```

Comment:
just flagging to confirm that `resource.id.id` is correct - I love when an id has an id!

Reply:
yes, it's kind of silly, but `resource.id` is a Valkyrie::ID or something like that, but instead of having a property called `value` or `val`, it's called `id` again. To add to the confusion in ways only Ruby can, the `Valkyrie::ID` object has a `to_s` method so you _can_ refer to `"#{resource.id}"` when it's being interpolated into a string. 🙃